src/sema/diagnostics.rs

```diff
@@ -5,48 +5,43 @@ use codespan_reporting::{diagnostic, files, term};
 use serde::Serialize;
 use std::{io, sync::Arc};
 
-/// Print the diagnostics to stderr with fancy formatting
-pub fn print_diagnostics(cache: &FileResolver, ns: &Namespace, debug: bool) {
-    let (files, file_id) = convert_files(ns, cache);
-
-    let writer = term::termcolor::StandardStream::stderr(term::termcolor::ColorChoice::Always);
-    let config = term::Config::default();
-
-    for msg in &ns.diagnostics {
-        if msg.level == Level::Debug && !debug {
-            continue;
-        }
-
-        let diagnostic = convert_diagnostic(msg, &file_id);
-
-        term::emit(&mut writer.lock(), &config, &files, &diagnostic).unwrap();
-    }
+#[derive(Default)]
+pub struct Diagnostics {
+    pub contents: Vec<Diagnostic>,
+    has_error: bool,
```

Comment:
one is pub and the other not?

Reply:
Both should be private. I forgot to remove the first `pub`. That was a lapse.